### ISSUE TITLE
Clarify license

### ISSUE THREAD

**Issue by lucaregini** [ikvmnet/ikvm#512]
It's pretty difficult to understand what is licensed under which license. Is the project Zlip / LPGL or GNU? Is it possible to chose any license or are there sections that are covered by one license rather than the other?

**Comment by lucaregini** [comment]
Is it possible to clarify clearly which portions are Zlib and which are not? Eventually by putting the right license in each top folder? 

 venerdì 12 aprile 2024 alle ore 18:53:23 CEST, Jerome Haltom ***@***.***> ha scritto: 
 
 


Different parts are covered by different licenses. The software is a composite. The portions of the software that are written by us are Zlib licensed. However, since a signficant part of the end result is OpenJDK, that part falls under their license, which is a variety of different things as well. Most of their code is GPLv2+CPE, though other portions are different.

**Comment by wasabii** [comment]
I think that's covered pretty well already. Is there something you can't find?

**Comment by lucaregini** [comment]
So to recap:
the folder openjdk/jdk8u GPLv2+CPE. Everything else is Zlib. At the same time the license here: https://github.com/ikvmnet/ikvm?tab=License-1-ov-file#readme doesn't specify any distinction and so yes I believe there is still something to clarify. I don't think a comment  in a github issue has the same value as a license file. 


    Il venerdì 12 aprile 2024 alle ore 20:15:08 CEST, Jerome Haltom ***@***.***> ha scritto:  
 
 


I think that's covered pretty well already. Is there something you can't find?

**Comment by wasabii** [comment]
Have you looked at the files?

**Comment by wasabii** [comment]
I mean some of the newer ones are missing headers it looks like. But the vast majority have headers.

**Comment by lucaregini** [comment]
Like this one for example  ikvm/src/IKVM.ByteCode/ByteCodeException.cs at main · ikvmnet/ikvm ?

| 
| 
| 
|  |  |

 |

 |
| 
|  | 
ikvm/src/IKVM.ByteCode/ByteCodeException.cs at main · ikvmnet/ikvm

A Java Virtual Machine and Bytecode-to-IL Converter for .NET - ikvmnet/ikvm
 |

 |

 |


Just code, no license.



    Il venerdì 12 aprile 2024 alle ore 20:27:39 CEST, Jerome Haltom ***@***.***> ha scritto:  
 
 


Have you looked at the files?

**Comment by lucaregini** [comment]
That's my point. The ones that do not have the header are under both GPL with exception and zlib.And yes I looked at the source code before raising this issue.My point is valid. 
    Il venerdì 12 aprile 2024 alle ore 20:31:28 CEST, Jerome Haltom ***@***.***> ha scritto:  
 
 


I mean some of the newer ones are missing headers it looks like. But the vast majority have headers.

**Comment by wasabii** [comment]
Yup. That's what I was asking. Looks like ByteCode is missing license information.

**Comment by lucaregini** [comment]
There are plenty of files that are missing the header. Do you need the complete list?
    Il venerdì 12 aprile 2024 alle ore 21:45:19 CEST, Jerome Haltom ***@***.***> ha scritto:  
 
 


Yup. That's what I was asking. Looks like ByteCode is missing license information.
